build.gradle

```diff
@@ -67,6 +67,22 @@ task nativeHeaders {
     }
 }
 
+def writeNativeVersionSources = tasks.register("writeNativeVersionSources", WriteNativeVersionSources) {
+    ["main", "shared", "curses"].each {
+        nativeSources.from("src/${it}/cpp")
+        nativeSources.from("src/${it}/headers")
+    }
```

Comment:
I wish we had source sets for C++ as well so we could reference them here instead of redeclaring the paths twice.

Reply:
I think there are, I just don't know where...